com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/util/NameUtil.java

```diff
@@ -89,6 +95,18 @@ public static String getNameWithProperLength(String name, int maxLength) {
         return name;
     }
 
+    private static String computeNamespacedNameWithLength(String name, String namespace, boolean applyNamespace, int maxLength) {
+        String prefix = "";
+        if (StringUtils.isNotEmpty(namespace) && applyNamespace) {
+            prefix = getNamespacePrefix(namespace);
+        }
+        return prefix + getNameWithProperLength(name, maxLength - prefix.length());
```

Comment:
if we add the prefix before - this could lead to the prefix also getting hashed for long prefixes; I agree the length exception is a bad way to notify the user though

Reply:
prefix is now passed as part of name so it will be possible to hash longer namespace-name strings but also hash the namespace itself